festim/hydrogen_transport_problem.py

```diff
@@ -126,22 +126,62 @@ def initialise(self):
     def defing_export_writers(self):
         """Defines the export writers of the model"""
         for export in self.exports:
-            # TODO implement when export.field is an int or str
-            # then find solution from index of species
+            # if name of species is given then replace with species object
+            for idx, field in enumerate(export.field):
+                if isinstance(field, str):
+                    export.field[idx] = F.find_species_from_name(field, self.species)
 
             if isinstance(export, (F.VTXExport, F.XDMFExport)):
                 export.define_writer(MPI.COMM_WORLD)
                 if isinstance(export, F.XDMFExport):
                     export.writer.write_mesh(self.mesh.mesh)
 
     def define_function_space(self):
-        elements = basix.ufl.element(
+        element_CG = basix.ufl.element(
             basix.ElementFamily.P,
             self.mesh.mesh.basix_cell(),
             1,
             basix.LagrangeVariant.equispaced,
         )
-        self.function_space = fem.FunctionSpace(self.mesh.mesh, elements)
+        elements = []
+        if len(self.species) > 1:
+            for spe in self.species:
+                if isinstance(spe, F.Species):
+                    # TODO check if mobile or immobile for traps
+                    elements.append(element_CG)
+            element_data = ufl.MixedElement(elements)
+        else:
+            element_data = element_CG
+
+        self.function_space = fem.FunctionSpace(self.mesh.mesh, element_data)
```

Comment:
I'd call this just `element,` `element_data` is kinda confusing. It _is_ an element object after all 
```suggestion
            element = ufl.MixedElement(elements)
        else:
            element = element_CG

        self.function_space = fem.FunctionSpace(self.mesh.mesh, element)
```